client/state/wordads/earnings/reducer.js

```diff
@@ -0,0 +1,39 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import {
+	WORDADS_EARNINGS_REQUEST,
+	WORDADS_EARNINGS_REQUEST_SUCCESS,
+	WORDADS_EARNINGS_REQUEST_FAILURE,
+} from 'state/action-types';
+import { combineReducers, createReducer } from 'state/utils';
+
+export const items = createReducer(
+	{},
+	{
+		[ WORDADS_EARNINGS_REQUEST_SUCCESS ]: ( state, action ) =>
+			Object.assign( {}, state, {
+				[ action.siteId ]: action.earnings,
+			} ),
+	}
+);
```

Comment:
Try to use the `keyedReducer` here so that you don't need to assign the `siteId` keys in state:
```js
items = keyedReducer( 'siteId', ( state, action ) => {
  switch ( action.type ) {
    case WORDADS_EARNINGS_REQUEST_SUCCESS:
      return action.earnings;
    default:
      return state;
  }
} );
```
`keyedReducer` will read the `siteId` property of the action and store the nested reducer's output into an `{ site1, site2, site3 }` object.